code/game/machinery/spaceheater.dm

```diff
@@ -138,34 +158,66 @@
 	if(cell)
 		cell.emp_act(severity)
 
+/obj/machinery/space_heater/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	default_unfasten_wrench(user, tool)
+	return TOOL_ACT_TOOLTYPE_SUCCESS
+
 /obj/machinery/space_heater/attackby(obj/item/I, mob/user, params)
 	add_fingerprint(user)
-	if(default_unfasten_wrench(user, I))
-		return
-	else if(istype(I, /obj/item/stock_parts/cell))
-		if(panel_open)
-			if(cell)
-				to_chat(user, "<span class='warning'>There is already a power cell inside!</span>")
-				return
-			else if(!user.transferItemToLoc(I, src))
-				return
-			cell = I
-			I.add_fingerprint(usr)
 
-			user.visible_message("<span class='notice'>\The [user] inserts a power cell into \the [src].</span>", "<span class='notice'>You insert the power cell into \the [src].</span>")
-			SStgui.update_uis(src)
-		else
-			to_chat(user, "<span class='warning'>The hatch must be open to insert a power cell!</span>")
-			return
-	else if(I.tool_behaviour == TOOL_SCREWDRIVER)
-		panel_open = !panel_open
+	if(default_deconstruction_screwdriver(user, icon_state, icon_state, I))
 		user.visible_message("<span class='notice'>\The [user] [panel_open ? "opens" : "closes"] the hatch on \the [src].</span>", "<span class='notice'>You [panel_open ? "open" : "close"] the hatch on \the [src].</span>")
-		update_icon()
-	else if(default_deconstruction_crowbar(I))
+		update_appearance()
+		return TRUE
+
+	if(default_deconstruction_crowbar(I))
+		return TRUE
+
+	if(istype(I, /obj/item/stock_parts/cell))
+		if(!panel_open)
+			balloon_alert(user, "Hatch must be open!")
+			return
+		if(cell)
+			balloon_alert(user, "Already a power cell inside!")
+			return
+		if(!user.transferItemToLoc(I, src))
+			return
+		cell = I
+		I.add_fingerprint(usr)
+		user.visible_message("<span class='notice'>\The [user] inserts a power cell into \the [src].</span>", "<span class ='notice'>You insert the power cell into \the [src].<span>")
+		SStgui.update_uis(src)
+		return TRUE
+	return ..()
+
+/obj/machinery/space_heater/CtrlClick(mob/user)
+	if(!can_interact(user))
 		return
+	toggle_power()
```

Comment:
I thought we were getting rid of this because it blocks pulling

Return ..() if not anchored, otherwise do this

Does it allow pulling when unanchored? If so it's fine

Reply:
yeah, im just gonna excise this function. It doesnt seem to want to work and its use is extremely limited anyways